packages/legend-studio-extension-query-builder/src/components/QueryBuilder_StudioPlugin.tsx

```diff
@@ -226,6 +234,41 @@ export class QueryBuilder_StudioPlugin
                   element,
                 );
                 queryBuilderExtension.queryBuilderState.resetData();
+                if (element.stereotypes.length !== 0) {
```

Comment:
Logic is not correct here. We should have a function called something like 
`isVersionedClass` that returns a boolean in legend-graph. Basically the logic is that a class is a versionedClass if it or any of its supertypes contain a stereotype in the `meta::pure::profiles.temporal` profile. This function will be reused by the work @YannanGao-gs is doing.